tests/stdlib/tstrutils.nim

```diff
@@ -817,5 +859,75 @@ bar
     doAssert s.endsWith('a') == false
     doAssert s.endsWith('\0') == false
 
+  when not defined(js) and not defined(i386):
```

Comment:
this is too coarse, some of those tests pass for js and (IIUC) for i386; instead just disable the parts that js can't handle so that test coverage stays good.
eg this works:
```
nim r -b:js --eval:'import strutils; echo parseHexInt("#FFFF")'
```
if needed, you can use whenRuntimeJs, whenVMorJs helpers from testutils

Reply:
fixed